spec/lib/pdf_filler/nj1040_pdf_spec.rb

```diff
@@ -180,6 +202,30 @@
           expect(pdf_fields["x  1000_2"]).to eq "0"
           expect(pdf_fields["undefined_9"]).to eq "0"
         end
+
+        it "fills pdf with Line 8 fields" do
+          expect(pdf_fields["Check Box43"]).to eq "Off"
+          expect(pdf_fields["Check Box44"]).to eq "Off"
+          expect(pdf_fields["x  1000_3"]).to eq "0"
+          expect(pdf_fields["undefined_10"]).to eq "0"
+        end
```

Comment:
[pebble] I was about to make the comment that this is the same test as the one on line 147. Then I looked into why they both existed, and I see that they exist indepedently under single filer vs married filing jointly. I'm a little confused about this breakdown of test organization. I feel like I'd much prefer to see tested logic grouped by feature (by line) because that's how I expect to read the tests when I'm looking for documentation on a specific behavior. Having to jump up and down the test file for blindness or other features and understand that it's in a filing status context seems confusing to me. Could we refactor this test file to remove "single filer" and "married filing jointly" as top-level contexts altogether and rather, test for those states _within_ specific lines as they are applicable? The fact that this test is duplicated in both places to me shows that these filing statuses are not always applicable to the outcome (like in this case) so I would rather organize by the topic that is affecting the test outcome (the line number and the specific exemption).

Reply:
@aloverso does this logic apply to the XML tests as well? I see that they're organized by trait rather than by feature right now, I can refactor those tests to be feature-based.